src/Spectre.libHeatmapDataScaling.Tests/ScalingDataUtilsTest.cpp

```diff
@@ -25,40 +25,48 @@ using namespace spectre::visualization;
 
 namespace
 {
-    class ScalingDataUtilsTest : public ::testing::Test
-    {
-    protected:
+	class ScalingDataUtilsTest : public ::testing::Test
+	{
+	protected:
         std::vector<double> testIntensities;
-        void SetUp() override
-       {
-           testIntensities = { 3.2, 2.69, 1.23, 1.23 };
-       }
-    };
+		void SetUp() override
+		{
+            testIntensities = { 3.2, 2.69, 1.23, 1.23 };
+		}
+	};
 
-    TEST_F(ScalingDataUtilsTest, returns_normalized_data)
-    {
-        auto result = normalizeData(testIntensities);
-        EXPECT_THAT(result, testing::ContainerEq(std::vector<double>{ 255.0, 188.98477157360404, 0.0, 0.0 }));
-    }
+	TEST_F(ScalingDataUtilsTest, returns_normalized_data)
+	{
+		auto result = scaleDataToRange(testIntensities);
+		std::vector<double> expected = { 255.0, 188.98, 0.0, 0.0 };
+		for (auto i = 0; i < 4; i++)
+		{
+            EXPECT_NEAR(result[i], expected[i], 2);
+		}
+	}
 
-    TEST_F(ScalingDataUtilsTest, returns_correct_data_size)
+	TEST_F(ScalingDataUtilsTest, returns_correct_data_size)
     {
-        EXPECT_EQ(normalizeData(testIntensities).size(), testIntensities.size());
-        EXPECT_EQ(normalizeData(testIntensities).size(), testIntensities.size());
-    }
+		EXPECT_EQ(scaleDataToRange(testIntensities).size(), testIntensities.size());
+		EXPECT_EQ(scaleDataToRange(testIntensities).size(), testIntensities.size());
```

Comment:
I don't understand: why are we calling the same function, with the same arguments twice? Do we expect machine code to evolve during the execution? 😄